### PR TITLE
add python

### PR DESCRIPTION
**Pull request by caishanli**
remove some not tested gpu about api in python
fix some name copy paste error
compile with lastest ncnn code

### CODE REVIEW

File: python/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+                shape.push_back(m.h);
+                shape.push_back(m.w);
+                strides.push_back(m.cstep * m.elemsize);
+                strides.push_back(m.w * m.elemsize);
+                strides.push_back(m.elemsize);
+            }
+            return py::buffer_info(
+                m.data,     /* Pointer to buffer */
+                m.elemsize, /* Size of one scalar */
+                format,     /* Python struct-style format descriptor */
```
Comment (nihui) [inline python/src/main.cpp:309]: elempack info lost
It seems numpy does not have the elempack semantic
what about another dimension for elempack != 1 ? [c, h, w, elempack]

File: python/src/pybind11_mat.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        format = pybind11::format_descriptor<float>::format();
+    }
+    if (m.elemsize == 2)
+    {
+        format = "e";
+    }
+    if (m.elemsize == 1)
+    { //int8 or uint8, so what?
+        format = pybind11::format_descriptor<int8_t>::format();
+    }
```
[the remark this answers is not in the document]
Comment (caishanli) [reply python/src/pybind11_mat.h:39]: "e" represent [python float16](https://docs.python.org/3/library/struct.html#format-characters)

File: python/setup.py.i [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import sys
+if sys.version_info < (3,0):
+  sys.exit('Sorry, Python < 3.0 is not supported')
+
+requirements = [
+    'numpy',
+    #'tqdm',
+    #'requests',
+    #'portalocker',
+    #'opencv-python'
```
[the remark this answers is not in the document]
Comment (caishanli) [reply python/setup.py.i:12]: this is for install with conda.
requirements can install with `pip install -U requirements.txt`  in readme.md

**Comment by nihui** [comment]
先合并了，以后改再pr就好

File: python/src/pybind11_mat.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+#include <pybind11/pybind11.h>
+
+#include <mat.h>
+
+std::string get_mat_format(const ncnn::Mat& m)
+{
+    std::string format;
+    if (m.elemsize == 4)
+    {
+        format = pybind11::format_descriptor<float>::format();
```
Comment (csukuangfj) [inline python/src/pybind11_mat.h:30]: What if it is of type `int32_t`?

**Pull request merged**